packages/wrangler/src/api/startDevWorker/bundle-allowed-paths.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * Browser DevTools will send `Network.loadNetworkResource` commands for source
+ * maps and source files. We only want to allow files referenced by the bundle
+ * to be requested (to prevent arbitrary file access). This module exports
+ * functions for checking whether a specific file may be requested as a source
+ * file, or source map (i.e. is safe to serve from the file system).
+ */
+import assert from "node:assert";
+import fs from "node:fs";
+import path from "node:path";
+import { fileURLToPath, pathToFileURL } from "node:url";
+import type { EsbuildBundle } from "../../dev/use-esbuild";
+import type { RawSourceMap } from "source-map";
+
+export function isAllowedSourcePath(
+	bundle: EsbuildBundle,
+	filePath: string
+): boolean {
+	const allowed = getBundleReferencedPaths(bundle);
+	return allowed.sourcePaths.has(filePath);
+}
+export function isAllowedSourceMapPath(
+	bundle: EsbuildBundle,
+	filePath: string
+): boolean {
+	const allowed = getBundleReferencedPaths(bundle);
+	return allowed.sourceMapPaths.has(filePath);
+}
```

Comment:
This is porting over the protections of arbitrary filesystem access

It's done differently to inspect.ts because, before, there were guarantees workerd were telling us about the files which were safe to request (which we kept in state mapped to IDs), whereas now that goes via a http port (Miniflare's loopback server) and anything with access could tell us arbitrary files were safe

Now we ensure any filepath requested is known from the bundle